Makefile

```diff
@@ -202,6 +202,7 @@ esp32:
 esp32-no-env: check-env check-esp32-env build/$(ESP32_CHIP)/firmware.envelope
 
 build/$(ESP32_CHIP)/firmware.envelope: build/$(ESP32_CHIP)/toit.bin
+build/$(ESP32_CHIP)/firmware.envelope: build/$(ESP32_CHIP)/system.snapshot
 build/$(ESP32_CHIP)/firmware.envelope: tools toit-tools
 	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/firmware.envelope create \
```

Comment:
I think this should eventually happen in the CMake files.
The Makefile should never be necessary, and just convenience to launch CMake.
Now both are necessary, which isn't great.

If I find the time I will try to move some of the critical things into cmake.